### DISCUSSION TITLE
syncronize multiply output

### DISCUSSION THREAD

**Discussion by Meinsda** [pschatzmann/arduino-audio-tools#123] [Q&A] [answered]
Is it possible that i can syncronize multipile stream player to play at the same time so i can make multiple room speaker?

**Comment by pschatzmann** [comment, marked as answer]
Well everything is possible. 

I think the easiest way to approach this is to use 1 ESP32 as Master which distributes the Audio via a shared protocol. As content you could use AAC or MP3 to limit the traffic on the network and as protocols UDP or ESP Now.

If you want to limit the traffic even further the Master would just send the command which url or which file to play...

**Reply by Meinsda** [reply]
Yes it's clear, i use one for a master and send my audio signal via broadcast to every client, but i need a system to syncronize all players that there is no time difference. Do you know a protocol?

**Comment by pschatzmann** [comment]
I would expect that if the speakers are in different rooms the small differences will not disturb.
However I have my doubt that it would even be possible to have them playing all the same sound up to the millisecond. 
But it would be interesting to see how big the differences really are...
